### PR TITLE
Added str tests in library

### PR DESCRIPTION
**Pull request by ayushmishra2005**
Added str tests in library  as a part of #76268

r? @matklad

### CODE REVIEW

File: library/alloc/tests/str.rs [hunk 1]
```diff
@@ -1921,3 +1921,11 @@ fn different_str_pattern_forwarding_lifetimes() {
...
     foo::<&str>("x");
 }
+
+#[test]
+fn test_str_concat() {
+    let a: String = "hello".to_string();
+    let b: String = "world".to_string();
+    let s: String = format!("{}{}", a, b);
+    assert_eq!(s.as_bytes()[9], 'd' as u8);
+}
```
Comment (jyn514) [inline library/alloc/tests/str.rs:1931]: This should go in `string.rs` instead I think. r=me after that

Comment (ayushmishra2005) [reply library/alloc/tests/str.rs:1931]: @jyn514 Shall I move in `string.rs` or wait for @matklad to provide any feedback?

Comment (jyn514) [reply library/alloc/tests/str.rs:1931]: Go ahead and move it.

Comment (ayushmishra2005) [reply library/alloc/tests/str.rs]: Done @jyn514 . Please check and approve

**Comment by jyn514** [comment]
@bors r+ rollup

Thanks for the PR!

**Pull request merged**